### PR TITLE
Add week view.

### PR DESCRIPTION
**Pull request by tclaar**
Add router for different views and use NavBar to change route.
Add context for calendar visibility and connect CalendarSelect to it.
Add WeekView component that contains events with basic fields.

### CODE REVIEW

**Review by Jasonsizemore24 — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,67 @@
 import './App.css';
+
+import { useState, createContext } from 'react';
+import { BrowserRouter, Routes, Route } from "react-router-dom"
```
Comment (b1gd3vd0g) [inline src/App.js:4]: Maybe I should have known this from class and havent watched enough, but is this just to change the URL when you click the buttons?

File: src/App.js [hunk 2]
```diff
@@ -1,13 +1,67 @@
...
+  // demo data
+  const calCtxValue = { 
+    '1': {
+      name: "Appointments",
+      events: [
+        // event list could be populated with event objects fetched using the event ID list 
+        // contained in the response from the calendar web service
+        {
+          name: "Dentist",
+          date_range: [ '2024-02-20', '2024-02-28' ],
```
Comment (b1gd3vd0g) [inline src/App.js:23]: A week long dentist appointment? Sick

File: src/components/NavBar.js [hunk 3]
```diff
@@ -1,21 +1,22 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
 
 function NavBar() {
   return (
     <footer className="navbar fixed-bottom navbar-light bg-light">
       <nav className="nav nav-fill w-100">
-        <a className="nav-link" href="#month">
+        <Link className="nav-link" to="/month/">
```
Comment (b1gd3vd0g) [inline src/components/NavBar.js:8]: I like this. I'm not familiar yet with the 'react-router-dom' but I think i get it.

**Review by b1gd3vd0g — APPROVED** [review]
Nice work

**Pull request merged**